python/pylibcudf/pylibcudf/io/parquet.pyx

```diff
@@ -221,3 +244,336 @@ cpdef read_parquet(
+        """
+        Sets the maximum uncompressed page size, in bytes.
+
+        Parameters
+        ----------
+        size_bytes : int
+            Maximum uncompressed page size, in bytes to set
+
+        Returns
+        -------
+        None
+        """
+        self.c_obj.set_max_page_size_bytes(size_bytes)
+
+    cpdef void set_max_page_size_rows(self, size_type size_rows):
+        """
+        Sets the maximum page size, in rows.
+
+        Parameters
+        ----------
+        size_rows : int
+            Maximum page size, in rows to set.
+
+        Returns
+        -------
+        None
+        """
+        self.c_obj.set_max_page_size_rows(size_rows)
+
+    cpdef void set_max_dictionary_size(self, size_t size_rows):
+        """
+        Sets the maximum dictionary size, in bytes.
+
+        Parameters
+        ----------
+        size_rows : int
+            Sets the maximum dictionary size, in bytes..
+
+        Returns
+        -------
+        None
+        """
+        self.c_obj.set_max_dictionary_size(size_rows)
+
+
+cdef class ParquetWriterOptionsBuilder:
+
+    cpdef ParquetWriterOptionsBuilder metadata(self, TableInputMetadata metadata):
+        """
+        Sets metadata.
+
+        Parameters
+        ----------
+        metadata : TableInputMetadata
+            Associated metadata
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.metadata(metadata.c_obj)
+        return self
+
+    cpdef ParquetWriterOptionsBuilder key_value_metadata(self, list metadata):
+        """
+        Sets Key-Value footer metadata.
+
+        Parameters
+        ----------
+        metadata : list[dict[str, str]]
+            Key-Value footer metadata
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.key_value_metadata(
+            [
+                {key.encode(): value.encode() for key, value in mapping.items()}
+                for mapping in metadata
+            ]
+        )
+        return self
+
+    cpdef ParquetWriterOptionsBuilder compression(self, compression_type compression):
+        """
+        Sets Key-Value footer metadata.
+
+        Parameters
+        ----------
+        compression : CompressionType
+            The compression type to use
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.compression(compression)
+        return self
+
+    cpdef ParquetWriterOptionsBuilder stats_level(self, statistics_freq sf):
+        """
+        Sets the level of statistics.
+
+        Parameters
+        ----------
+        sf : StatisticsFreq
+            Level of statistics requested in the output file
+
+        Returns
+        -------
+        Self
+        """
+        self.c_obj.stats_level(sf)
+        return self
+
+    cpdef ParquetWriterOptionsBuilder int96_timestamps(self, bool enabled):
+        """
+        Sets whether int96 timestamps are written or not.
```

Comment:
This is a bit unclear. If false, this still writes the data, just as a different type. Look at the C++ code to see what this should say.